### PR TITLE
SSR2: more improvements

### PR DESCRIPTION
**Pull request by Popov72**
Add new features and fix some bugs.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13631/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13631/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13631/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13631/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13631/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13631/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13631/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13631/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13631/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13631/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Engines/engineFeatures.ts [hunk 1]
```diff
@@ -42,6 +42,9 @@ export interface EngineFeatures {
     /** Indicates that SSAO2 is supported */
     supportSSAO2: boolean;
 
+    /** Indicates that SSR Rendering Pipeline is supported */
+    supportSSR: boolean;
```
Comment (sebavan) [inline packages/dev/core/src/Engines/engineFeatures.ts:46]: Should it be based on lower level features instead ? as SSR is not an engine feature I am afraid of the flag multiplication ?

File: packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssrRenderingPipeline.ts [hunk 2]
```diff
@@ -841,9 +996,9 @@ export class SSRRenderingPipeline extends PostProcessRenderPipeline {
         this._blurPostProcessX = new PostProcess(
             "SSRblurX",
             "screenSpaceReflection2Blur",
-            ["blurQuality", "texelOffsetScale"],
+            ["texelOffsetScale"],
             ["textureSampler"],
-            1,
+            this._useBlur() ? 1 / (this._ssrDownsample + 1) : 1,
```
Comment (sebavan) [inline packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssrRenderingPipeline.ts:1001]: should all the size params be handled in resize as well ?

Comment (Popov72) [reply packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssrRenderingPipeline.ts:1008]: Not sure I understand. Resizing is handled by the post process base class, and textures are recreated with the correct ratio value.

**Review by sebavan — APPROVED** [review]

**Pull request merged**